webdriver-extension/explainer.md

```diff
@@ -0,0 +1,106 @@
+
+### Goals
+
+ - Allow user agents to create mock sensor devices that have certain characteristics and behave in a specific way.
+ - Easy controlling mock sensor via [WebDriver Protocol Extension Commands](https://w3c.github.io/webdriver/#protocol-extensions).
+
+### Non-goals
+
+ - Not intend to expose Mock Sensor API to web developer directly.
+
+### Draft Spec
+
+[Draft Spec](https://w3c.github.io/sensors/webdriver-extension)
+
+
+## What is Mock Sensor?
+
+A mock sensor simulates the behavior of a [platform sensor](https://w3c.github.io/sensors/#concept-platform-sensor) in controlled ways for the purpose of sufficiently meeting requirements of automation testing for Generic Sensor API. A mock sensor should have the following capabilities:
+ - Reporting mock sensor readings, which is a source of mocking information about the environment, to [Sensor](https://w3c.github.io/sensors/#sensor) objects at the rate of its requested sampling frequency if the user agent is [allowed to expose sensor readings](https://w3c.github.io/sensors/#can-expose-sensor-readings). 
+ - User specified mock sensor reading, user agent allows user-customized mock sensor reading for a mock sensor.
+ - A sampling frequency, which is defined as a frequency at which the user agent obtains sensor readings from mock sensor. The sampling frequency has an upper and lower bounds, which is designed for the purpose of calculating expected sampling frequency. For instance, when construct a Sensor object with frequency option that exceed the maximium supported sampling frequency, the exact requested sampling frequency should capped to the maximum sampling frequency.
+ - A requested sampling frequency, used for calculating expected sampling frequency.
+ - A connection flag that is used for switching the connection between Sensor object and mock sensor, for testing the `NotReadableError` exception when calling [`sensor.start()`](https://w3c.github.io/sensors/#sensor-start) fails.
+
+## Getting started
+
+### Basic Flow
+
+This is a basic work flow that demonstrates how WebDriver interacts with browsers and tests that run on a HTTP client.
+
+Tests could be written in any languages, as long as it wraps the WebDriver HTTP APIs. Basically, tests send commands across the common WebDriver API with [WebDriver wire protocol](https://w3c.github.io/webdriver/#protocol), on the other side, the WebDriver, which runs an HTTP server that implements the extension APIs in the WebDriver spec, receives those commands and execute them on the actual browser to manipulate mock sensors and then returns the result all the way back.
+
+![WebDriver Extension Command Flow Diagram](./images/flow_diagram.png)
+
+### Create mock sensor
+
+ To create an "accelerometer" mock sensor with session ID 21, the local end would send a `POST` request to `/session/21/sensor/` with the body:

```

Comment:
By convention, WebDriver endpoint URL shouldn't end with a trailing `/`, so `/session/21/sensor` would be more appropriate. Similarly for other endpoint URLs in this PR.